StyleRuleCSSStyleDeclaration should use RefCounted instead of re-implementing the same thing

achristensen07 opened the pull request:
#### 35296e904fb2aa582e24b156559458cdd04a77e0
<pre>
StyleRuleCSSStyleDeclaration should use RefCounted instead of re-implementing the same thing
<a href="https://bugs.webkit.org/show_bug.cgi?id=258087">https://bugs.webkit.org/show_bug.cgi?id=258087</a>
rdar://110795757

Reviewed by Chris Dumez.

Do the same thing with a few other classes.
No change in behavior.

* Source/WebCore/css/PropertySetCSSStyleDeclaration.cpp:
(WebCore::StyleRuleCSSStyleDeclaration::StyleRuleCSSStyleDeclaration):
(WebCore::StyleRuleCSSStyleDeclaration::ref): Deleted.
(WebCore::StyleRuleCSSStyleDeclaration::deref): Deleted.
* Source/WebCore/css/PropertySetCSSStyleDeclaration.h:

Canonical link: <a href="https://commits.webkit.org/266165@main">https://commits.webkit.org/266165@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/05218b932cc6018683c6a75557cab325d283c763

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13032 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13350 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13681 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14770 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12421 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13100 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15856 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13376 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15116 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13199 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13891 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11013 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15223 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11179 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11768 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18836 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12254 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11934 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15149 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12417 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10302 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11685 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16003 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1483 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12260 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/css/PropertySetCSSStyleDeclaration.h:
```diff
@@ -100,9 +100,9 @@ class StyleRuleCSSStyleDeclaration final : public PropertySetCSSStyleDeclaration
-    void ref() final;
-    void deref() final;
+
+    void ref() final { RefCounted<StyleRuleCSSStyleDeclaration>::ref(); }
```
Not sure `<StyleRuleCSSStyleDeclaration>` is really needed nowadays.

cdumez commented on Source/WebCore/css/PropertySetCSSStyleDeclaration.h:
```diff
@@ -100,9 +100,9 @@ class StyleRuleCSSStyleDeclaration final : public PropertySetCSSStyleDeclaration
-    void deref() final;
+
+    void ref() final { RefCounted<StyleRuleCSSStyleDeclaration>::ref(); }
+    void deref() final { RefCounted<StyleRuleCSSStyleDeclaration>::deref(); }
```
ditto.

cdumez reviewed (approved):
LGTM

cdumez commented on Source/WebCore/css/CSSRuleList.h:
```diff
@@ -47,31 +48,31 @@ class CSSRuleList {
 // The rule owns the live list.
 template <class Rule>
 class LiveCSSRuleList final : public CSSRuleList {
+    WTF_MAKE_FAST_ALLOCATED;
```
CSSRuleList has it so not needed.

achristensen07 replied:
CSSRuleList used to have it, but I had to remove it to make StaticCSSRuleList RefCounted, so I moved it to all other CSSRuleList inheritors.

cdumez replied:
Ah ok, i missed that

cdumez commented on Source/WebCore/css/CSSStyleSheet.cpp:
```diff
@@ -58,6 +58,7 @@ static Style::Scope& styleScopeFor(ContainerNode& treeScope)
 }
 
 class StyleSheetCSSRuleList final : public CSSRuleList {
+    WTF_MAKE_FAST_ALLOCATED;
```
CSSRuleList has it so not needed.

cdumez reviewed (approved):
still LGTM

achristensen07 committed:
StyleRuleCSSStyleDeclaration should use RefCounted instead of re-implementing the same thing
https://bugs.webkit.org/show_bug.cgi?id=258087
rdar://110795757

Reviewed by Chris Dumez.

Do the same thing with a few other classes.
No change in behavior.

* Source/WebCore/css/PropertySetCSSStyleDeclaration.cpp:
(WebCore::StyleRuleCSSStyleDeclaration::StyleRuleCSSStyleDeclaration):
(WebCore::StyleRuleCSSStyleDeclaration::ref): Deleted.
(WebCore::StyleRuleCSSStyleDeclaration::deref): Deleted.
* Source/WebCore/css/PropertySetCSSStyleDeclaration.h:

Canonical link: https://commits.webkit.org/266165@main
